src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,139 @@
+            head = head.previous;
+        } else {
+            Node<T> validNode = getNodeByIndex(index);
+            Node<T> node = new Node<T>(validNode.previous, value, validNode);
+            validNode.previous = node;
+            node.next.previous = node;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNodeByIndex(index);
+        T oldVal = node.value;
+        node.value = value;
+        return oldVal;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> removeTarget = getNodeByIndex(index);
+        unlink(removeTarget);
+        return removeTarget.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> node = getNodeByValue(object);
+        if (node == null) {
+            return false;
+        }
+        unlink(node);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<T> {
+        private Node<T> previous;
+        private T value;
+        private Node<T> next;
+
+        public Node(Node<T> previous, T value, Node<T> next) {
+            this.previous = previous;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private void throwException(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Index out of bounds for length");
+        }
+    }
+
+    private Node<T> getNodeByIndex(int targetIndex) {
+        throwException(targetIndex);
+        Node<T> currentNode = tail;
+        for (int i = size - 1; i != targetIndex; i--) {
+            currentNode = currentNode.previous;
+        }
+        return currentNode;
+    }
+
+    private Node<T> getNodeByValue(T value) {
+        Node<T> newNode = head;
+        while (newNode != null) {
+            if (newNode.value == value || newNode.value != null && newNode.value.equals(value)) {
+                return newNode;
+            }
+            newNode = newNode.next;
+        }
+        return null;
+    }
+
+    private void unlink(Node<T> removeNode) {
+        Node<T> nodePrev = removeNode.previous;
+        Node<T> nodeNext = removeNode.next;
+        if (nodePrev == null) {
+            head = nodeNext;
+        } else {
+            nodePrev.next = nodeNext;
+        }
+        if (nodeNext == null) {
+            tail = nodePrev;
+        } else {
+            nodeNext.previous = nodePrev;
+        }
+        size--;
```

Comment:
Extract all methods for modification of the linked list to MyLinkedList class. This way, you can make classNode<T> non-static.

Reply:
I don't quite understand what you mean and why i must do it, let's see what the mentor will say.